File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/holder/CommentHolder.java
package com.module.playways.room.room.comment.holder;

import android.graphics.drawable.Drawable;
import android.support.v7.widget.RecyclerView;
import android.text.SpannableStringBuilder;
import android.text.TextUtils;
import android.view.View;

import com.common.core.avatar.AvatarUtils;
import com.common.core.userinfo.model.UserLevelType;
import com.common.image.fresco.BaseImageView;
import com.common.utils.SpanUtils;
import com.common.utils.U;
import com.common.view.ex.ExTextView;
import com.component.busilib.view.AvatarView;
import com.component.level.utils.LevelConfigUtils;
import com.module.playways.room.room.comment.adapter.CommentAdapter;
import com.module.playways.room.room.comment.listener.CommentViewItemListener;
import com.module.playways.room.room.comment.model.CommentModel;
import com.module.playways.R;

public class CommentHolder extends RecyclerView.ViewHolder {
    public final String TAG = "CommentHolder";

    AvatarView mAvatarIv;
    ExTextView mCommentTv;

    CommentModel mCommentModel;
    int mPostion;

    public CommentHolder(View itemView, CommentAdapter.CommentAdapterListener mCommentItemListener) {
        super(itemView);

        mAvatarIv = itemView.findViewById(R.id.avatar_iv);
        mCommentTv = itemView.findViewById(R.id.comment_tv);
        mAvatarIv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mCommentItemListener != null && mCommentModel != null && mCommentModel.getFakeUserInfo() == null) {
                    mCommentItemListener.clickAvatar(mCommentModel.getUserInfo().getUserId());
                }
            }
        });
    }

    public void bind(int position, CommentModel model) {
        mPostion = position;
        mCommentModel = model;

        mAvatarIv.bindData(model.getUserInfo());

        // 为了保证书写从左到右
        SpanUtils spanUtils = new SpanUtils().append("\u202D");
        if (model.getUserInfo() != null
                && model.getUserInfo().getRanking() != null
                && LevelConfigUtils.getSmallImageResoucesLevel(model.getUserInfo().getRanking().getMainRanking()) > 0) {
            Drawable drawable = U.getDrawable(LevelConfigUtils.getSmallImageResoucesLevel(model.getUserInfo().getRanking().getMainRanking()));
            drawable.setBounds(0, 0, U.getDisplayUtils().dip2px(22), U.getDisplayUtils().dip2px(19));
            spanUtils.appendImage(drawable, SpanUtils.ALIGN_CENTER);
        }
        if (!TextUtils.isEmpty(model.getNameBuilder())) {
            spanUtils.append(model.getNameBuilder());
        }

        if (model.getUserInfo().getHonorInfo() != null && model.getUserInfo().getHonorInfo().isHonor()) {
            Drawable honorDrawable = U.getDrawable(R.drawable.person_honor_icon);
            honorDrawable.setBounds(0, 0, U.getDisplayUtils().dip2px(23), U.getDisplayUtils().dip2px(14));
            spanUtils.appendImage(honorDrawable, SpanUtils.ALIGN_CENTER).append(" ");
        }

        if (!TextUtils.isEmpty(model.getStringBuilder())) {
            spanUtils.append(model.getStringBuilder());
        }
        spanUtils.append("\u202C");
        mCommentTv.setText(spanUtils.create());
    }
}

File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/holder/CommentHolder.kt
package com.module.playways.room.room.comment.holder


import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.View
import com.common.core.myinfo.MyUserInfoManager

import com.common.utils.SpanUtils
import com.common.utils.U
import com.common.view.ex.ExTextView
import com.component.busilib.view.AvatarView
import com.component.level.utils.LevelConfigUtils
import com.module.playways.room.room.comment.adapter.CommentAdapter
import com.module.playways.room.room.comment.model.CommentModel
import com.module.playways.R

class CommentHolder(itemView: View, mCommentItemListener: CommentAdapter.CommentAdapterListener?) : RecyclerView.ViewHolder(itemView) {
    val TAG = "CommentHolder"

    private val mAvatarIv: AvatarView = itemView.findViewById(R.id.avatar_iv)
    private val mCommentTv: ExTextView = itemView.findViewById(R.id.comment_tv)

    private var mCommentModel: CommentModel? = null
    private var mPostion: Int = 0

    init {
        mAvatarIv.setOnClickListener {
            if (mCommentModel?.isFake == false) {
                mCommentModel?.userInfo?.userId?.let {
                    mCommentItemListener?.clickAvatar(it)
                }
            }
        }
    }

    fun bind(position: Int, model: CommentModel) {
        mPostion = position
        mCommentModel = model

        if (model.isFake) {
            mAvatarIv.bindData(model.userInfo, model.fakeUserInfo?.nickName, model.fakeUserInfo?.avatarUrl)
        } else {
            mAvatarIv.bindData(model.userInfo, model.fakeUserInfo?.nickName, model.userInfo?.avatar)
        }

        // 为了保证书写从左到右
        val spanUtils = SpanUtils().append("\u202D")
        if (model.userInfo != null
                && model.userInfo!!.ranking != null
                && LevelConfigUtils.getSmallImageResoucesLevel(model.userInfo!!.ranking.mainRanking) > 0) {
            val drawable = U.getDrawable(LevelConfigUtils.getSmallImageResoucesLevel(model.userInfo!!.ranking.mainRanking))
            drawable.setBounds(0, 0, U.getDisplayUtils().dip2px(22f), U.getDisplayUtils().dip2px(19f))
            spanUtils.appendImage(drawable, SpanUtils.ALIGN_CENTER)
        }
        if (!TextUtils.isEmpty(model.nameBuilder)) {
            spanUtils.append(model.nameBuilder!!)
        }

        if (model.userInfo!!.honorInfo != null && model.userInfo!!.honorInfo.isHonor()) {
            val honorDrawable = U.getDrawable(R.drawable.person_honor_icon)
            honorDrawable.setBounds(0, 0, U.getDisplayUtils().dip2px(23f), U.getDisplayUtils().dip2px(14f))
            spanUtils.appendImage(honorDrawable, SpanUtils.ALIGN_CENTER).append(" ")
        }

        if (!TextUtils.isEmpty(model.stringBuilder)) {
            spanUtils.append(model.stringBuilder!!)
        }
        spanUtils.append("\u202C")
        mCommentTv.text = spanUtils.create()
    }
}
